extensions/amp-story/1.0/utils.js

```diff
@@ -231,6 +231,30 @@ export function getSourceOriginForElement(element, url) {
   return domainName;
 }
 
+/**
+ * Checks if document is being served from cache.
+ * @param {!Document} doc
+ * @return {boolean}
+ */
+export function isServedFromCache(doc) {
+  const urlService = Services.urlForDoc(doc.body);
+  const url = doc.location.href;
+  const sourceOrigin = getSourceOrigin(url);
+  const {origin} = urlService.parse(url);
+
+  return sourceOrigin !== origin;
+}
+
+/**
+ * Checks if url is absolute.
+ * @param {string} url
+ * @return {boolean}
+ */
+export function isAbsoluteUrl(url) {
```

Comment:
This check looks a bit more robust: https://github.com/ampproject/amphtml/blob/master/src/url.js#L594